app/helpers/decorator_helper.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module DecoratorHelper
+
+  # This smells of :reek:NilCheck
+  def current_account
+    @_decorated_current_account ||= controller.current_account&.decorate
```

Comment:
As @hallelujah mentioned, this could be:
```suggestion
    @_decorated_current_account ||= super&.decorate
```

Then I can make the other method be protected again 😄